### DISCUSSION TITLE
EvaluateExpression

### DISCUSSION THREAD

**Discussion by kcap** [alpinejs/alpine#4277] [7. Bugs (archived)] [answered]
Hello
I noticed that the evaluate function in the directive can take an object as an argument
`<div x-my-directive="{option1, option2}">`
evaluate(expression) returns {option1: value, option2: value}
but if there is a dot in variable name 
`<div x-my-directive="{option.test1, option.test2}">`
it returns error :  "Alpine Expression Error: Unexpected token '.'"

**Comment by kcap** [comment]
ok i can use template literals
`x-my-directive="`${option.test1}|${option.test2}`"`
and parse it 
```
value1 = evaluated.split("|")[0]
value2 = evaluated.split("|")[1]
```
but this doesn't look like good code

**Comment by ekwoka** [comment, marked as answer]
This is just javascript.

evaluate evaluates it as javascript

if you did

```js
const data = { option.test1, option.test2 }
```

that would be a syntax error.

If you want too values like that, use an array/tuple

```html
<div x-directive="[option.test1,option.test2]">
```

then you can just get it like

```js
evaluate(([test1, test2]) => {})
```

or whatever.

**Reply by kcap** [reply]
thank you
